app/src/client/tests/UserProperty.test.tsx

```diff
@@ -0,0 +1,261 @@
+import React from 'react';
+import { describe, it, expect, vi, beforeEach, Mock } from 'vitest';
+import { screen, fireEvent, waitFor } from '@testing-library/react';
+
+import { UserProperty } from '../components/buildPage/UserProperty';
+import { renderInContext } from 'wasp/client/test';
+import { useQuery } from 'wasp/client/operations';
+
+// Mock the wasp/client/operations
+vi.mock('wasp/client/operations', () => ({
+  useQuery: vi.fn(() => ({
+    // todo: fix the data strucure. pass the user added models
+    data: [
+      { uuid: '1', type_name: 'testProperty', model_name: 'Model1', json_str: {} },
+      { uuid: '2', type_name: 'testProperty', model_name: 'Model2', json_str: {} },
+    ],
+    isLoading: false,
+    refetch: vi.fn(),
+  })),
+  getModels: vi.fn(),
+}));
+
+describe('UserProperty', () => {
+  const mockProps = {
```

Comment:
maybe this json schema could just be moved to separate file since it is on the long side, a a bit disrupts reading flow of the tests.

Reply:
The schemas related to the tests are now moved to a new file called test/mocks.ts